File: 2017/24/spec/runner.spec.js
import { parse } from "../lib/parser";
import { part1, part2 } from "../lib/runner";

const INPUT = ` 0/2
    2/2
    2/3
    3/4
    3/5
    0/1
    10/1
    9/10 `
describe("Runner", () => {
  test("part 1 passes tests cases", () => {
    let parsed = parse(INPUT)
    expect(part1(parsed)).toEqual(31)
  });

  test("part 2 passes tests cases", () => {
  });
});

File: 2020/common/spec/runner_spec.rb
# frozen_string_literal: true

require "runner"
require "parser"

describe Runner do
  it "calculates the correct answer for the test input for part1" do
    expect(Runner.part_1(Parser.parse(sample_input))).to eq(514_579)
  end

  it "calculates the correct answer for the test input part2" do
    expect(Runner.part_2(Parser.parse(sample_input))).to eq(241_861_950)
  end

  def sample_input
    <<~INPUT.strip
      1721
      979
      366
      299
      675
      1456
    INPUT
  end
end
